### DISCUSSION TITLE
What happens if node is reached during an election

### DISCUSSION THREAD

**Discussion by natilivni** [dotnet/dotNext#47] [Q&A]
Hello,

I was wondering how to handle cases in which a node is reached during an election in which case the leader is null. Is there a way to wait in code for an election to complete and then proceed or is the only way to handle this is through client retries until the cluster has elected a leader?

**Comment by sakno** [comment]
Depends on characteristics of your distributed system. If your system is AP then reading is allowed. Otherwise, the node should not perform read/write if leader is not visible to it. Moreover, read request must be processed by the leader too if you building a system with strong consistency instead of eventual consistency.

Waiting is not possible because the node don't know when the leader will be elected. Probably, the node is in partitioned segment of the network and can stay in this state for a long time.
